### PR TITLE
new-plot import

### PR DESCRIPTION
**Pull request by larkin**
Imports the new plot and removes the old plot.  Has been reworked multiple times...

### CODE REVIEW

File: src/api/telemetry/TelemetryMetadataManager.js [hunk 1]
```diff
@@ -78,6 +78,18 @@ define([
         valueMetadata.source = valueMetadata.source || valueMetadata.key;
         valueMetadata.hints = valueMetadata.hints || {};
 
+        if (valueMetadata.format == 'enum') {
```
Comment (ivikash) [inline src/api/telemetry/TelemetryMetadataManager.js]: @larkin I think this should be `===` instead of `==`

File: src/plugins/plot/src/chart/MCTChartLineStepAfter.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        startIndexForPointAtIndex: function (index) {
+            if (index === 0) {
+                return 0;
+            }
+            return 2 + ((index - 1) * 4);
+        },
+        addPoint: function (point, start, count) {
+            if (start === 0 && this.count === 0) {
+                // First point is easy.
+                this.buffer[start] = point.x;
```
Comment (ivikash) [inline src/plugins/plot/src/chart/MCTChartLineStepAfter.js]: All the `if..else` have `this.buffer[start] = point.x;` part common. So moving it out of `if...else` would be a bad idea?

File: index.html [hunk 3]
```diff
@@ -39,7 +39,7 @@
             );
             openmct.install(openmct.plugins.MyItems());
             openmct.install(openmct.plugins.LocalStorage());
-            openmct.install(openmct.plugins.Espresso());
+            openmct.install(openmct.plugins.Snow());
```
Comment (VWoeltjen) [inline index.html]: Was this change intentional?

**Comment by akhenry** [comment]
Per discussion between @larkin and @akhenry make plot not require same range key for all series.

**Comment by charlesh88** [comment]
### Testing notes (Cliff Notes version)

1. Make new Overlay Plot. Add telem to it. 
1. Try zooming, panning, etc.
1. Change view options in the Inspector. Does stuff change as expected?
1. Save your new thing. 
1. Test it in Browse mode. Toggle the legend to expand and contract it. Zoom, pan, etc.
1. Add it to a Layout. Do actions from the previous step.
1. Does it all work as expected? If not, file issues.
1. Same as above for a new Stacked Plot.

**Comment by akhenry** [comment]
Unable to change plot line color

To reproduce:
1. Create a new overlay plot
2. Drag in a telemetry point (eg. warp:nss.sn)
3. Under "series options" click on the Color select dropdown
4. Dropdown box does not appear, and user is unable to change series color.

**Comment by charlesh88** [comment]
Just pushed up my work to fix palette problem, referenced here above by Andrew and in #1832. Plots, FP, Summary Widgets and Style Guide example palettes should all be looking correct now.

**Comment by larkin** [comment]
@akhenry @VWoeltjen I would like to propose temporarily removing our testing requirement for this plugin for a short while.  I'm very interested in removing Angular from this plugin in the future and in order to do that I'd have to rewrite a lot of tests.  We've also done a lot of testing of this, and having it not present in master branch is becoming more of a hassle than a benefit.  And there is a pile of issues related to plots that will close when we merge it.

In order to get this mergeable, I can open an issue for backfill of tests, and resolve all style and build-breaking issues in this branch.  Let me know if that's okay with you both.

**Comment by larkin** [comment]
@akhenry @VWoeltjen rebased, styles fixed and test issues resolved.  This is ready for review and merge.

# Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? N, filed https://github.com/nasa/openmct/issues/1920 for backfill
3. Command line build passes? Y
4. Changes have been smoke-tested? Y

**Comment by akhenry** [comment]
Plot series color preference does not appear to be being persisted right now. Or, it is persisted unpredictably.

I have seen this manifest in two subtly different ways. Both easy to reproduce with SWG:
1. Setting line color on create. Create a new plot, drop in a SWG, select a line color for the series, Save As. Line color reverts immediately to default color.
2. Setting line color when editing an existing object. Edit a plot, select a line color for the series, save and finish. New line color is correctly displayed in browsed object. Reload the page, line color reverts to default.

File: platform/telemetry/src/TelemetryCapability.js [hunk 4]
```diff
@@ -205,6 +205,9 @@ define(
                 },
                 getPointCount: function () {
                     return telemetry.length;
+                },
+                getData: function () {
```
[the remark this answers is not in the document]
Comment (larkin) [reply platform/telemetry/src/TelemetryCapability.js]: Used in VISTA for exporting data.

File: src/plugins/plot/src/telemetry/PlotController.js [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+    PlotController.prototype.loadMoreData = function (range, purge) {
+        this.config.series.map(function (plotSeries) {
+            this.startLoading();
+            plotSeries.load({
+                    size: this.$element[0].offsetWidth,
+                    start: range.min,
+                    end: range.max
+                })
+                .then(this.stopLoading.bind(this));
+            if (purge) {
```
[the remark this answers is not in the document]
Comment (larkin) [reply src/plugins/plot/src/telemetry/PlotController.js]: This is to purge data outside the current visible range on time conductor adjustments.  I don't believe that is done elsewhere.  Hrm, I see some unused code related to `resetOnAppend` that I will remove.

[the remark this answers is not in the document]
Comment (akhenry) [reply src/plugins/plot/src/telemetry/PlotController.js]: Ah, I see. I thought `.load` replaced data with the newly fetched data, which would be time bound by the server, but I see in [PlotSeries.fetch that it's merged with the existing data](https://github.com/nasa/openmct/blob/204f6be9d0ad7b82add248dc46f31efb72cf3592/src/plugins/plot/src/configuration/PlotSeries.js#L145). It seems as though in the case of a conductor bounds change, the merging (which includes sorting etc.) is redundant? The data could just be replaced with the requested telemetry returned by the Telemetry API?

[the remark this answers is not in the document]
Comment (larkin) [reply src/plugins/plot/src/telemetry/PlotController.js]: @akhenry theoretically, yes, it could be replaced completely with historical telem returned by the telem API.  However, if there is any delay between realtime feed and historical ingestion, or if the historical request takes long enough to return, then there could be a race resulting in a gap between historical and realtime streams.  I prefer to merge the data to prevent this from occurring.

**Comment by larkin** [comment]
@akhenry have resolved color persistence and a bug with interpolation persistence.  Verified all other options work as expected.

Also removed a few extra debugging statements and some unused code.  Give it another review and let me know if anything needs change.

**Comment by akhenry** [comment]
Changes look good. There is a non-blocking bug where alarm lines remain present in a plot after the associated object has been removed, but this will be filed separately.

## Reviewer Checklist

* Changes appear to address issue? Y
* Appropriate unit tests included? N/A - Followup issue #1920 to address this
* Code style and in-line documentation are appropriate? Y
* Commit messages meet standards? Y
